packages/transaction-controller/src/utils/batch.ts

```diff
@@ -364,7 +376,23 @@ async function addTransactionBatchWithHook(
     throw new Error('No publish batch hook provided');
   }
 
+  const chainId = getChainId(networkClientId);
   const batchId = generateBatchId();
+
+  if (requireApproval) {
+    const txBatchMeta: TransactionBatchMeta = {
```

Comment:
Assuming this will grow, should we create a `newBatchMetadata` function to just return the object?

Reply:
Created `newBatchMetadata` function.